### PR TITLE
cloud availability updater: automatic PR creation

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/3777
We want to automatically open PR on our internal platform repo for each connector eligible for Cloud.

## How
* Implement calls to GitHub API to create a PR and check if one already exists for the same branch.
* Implement `deploy_eligible_connectors_to_cloud_repo` which:
  1. Clones the internal platform repo to a tmp dir
  2. Iterate on eligible connectors to deploy these to cloud (the final step of the deployment is actually a PR creation on the platform internal repo)
 * Provide the `run-qa-engine` github action workflow with the required env var to perform PR creation.
 * Disable the automatic scheduling of this github action workflow. We'll turn it on again once all QA checks are implemented.
 
An example of the automatically created PR is available [here](https://github.com/airbytehq/airbyte-platform-internal/pull/4388/files#diff-e74bb09793c76832ed4fbfdb0f933dd782d3f4ae23d93adb4dadc0a3fc831f51).

## Recommended reading order
1. https://github.com/airbytehq/airbyte/blob/97b34efcc8e5943f7d9ac1d067c88c476e2847fa/tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py#L145
2. https://github.com/airbytehq/airbyte/blob/97b34efcc8e5943f7d9ac1d067c88c476e2847fa/tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py#L121
3. https://github.com/airbytehq/airbyte/blob/97b34efcc8e5943f7d9ac1d067c88c476e2847fa/tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py#L97
4. https://github.com/airbytehq/airbyte/blob/97b34efcc8e5943f7d9ac1d067c88c476e2847fa/tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py#L87
5. https://github.com/airbytehq/airbyte/blob/97b34efcc8e5943f7d9ac1d067c88c476e2847fa/.github/workflows/run-qa-engine.yml#L1

## 🚨 User Impact 🚨
None as the Run QA Engine workflow scheduling is disabled.

### CODE REVIEW

**Comment by evantahler** [comment]
@alafanechere is there an example PR you have made that demonstrates this? That's the unit test for this PR :D

File: .github/workflows/run-qa-engine.yml [hunk 1]
```diff
@@ -2,10 +2,10 @@ name: Run QA Engine
...
 on:
   workflow_dispatch:
-  schedule:
-    # 1pm UTC is 6am PDT.
-    # same time as Generate Build Report
-    - cron: "0 13 * * *"
+  # schedule:
+  ## 1pm UTC is 6am PDT.
+  ## same time as Generate Build Report
+  # - cron: "0 13 * * *"
```
Comment (evantahler) [inline .github/workflows/run-qa-engine.yml:8]: 👍 disabled for now

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py [hunk 2]
```diff
@@ -1,47 +1,58 @@
 #
-# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
```
Comment (evantahler) [inline tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py:2]: I think it's 2023...

**Comment by alafanechere** [comment]
@evantahler yep, this PR https://github.com/airbytehq/airbyte-platform-internal/pull/4388

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py [hunk 3]
```diff
@@ -1,47 +1,58 @@
...
-logger.setLevel(logging.INFO)
 
 
 def clone_airbyte_cloud_repo(local_repo_path: Path) -> git.Repo:
-    logging.info(f"Cloning {AIRBYTE_CLOUD_GITHUB_REPO_URL} to {local_repo_path}")
-    return git.Repo.clone_from(AIRBYTE_CLOUD_GITHUB_REPO_URL, local_repo_path, branch=AIRBYTE_CLOUD_MAIN_BRANCH_NAME)
+    logger.info(f"Cloning {AIRBYTE_PLATFORM_INTERNAL_GITHUB_REPO_URL} to {local_repo_path}")
+    return git.Repo.clone_from(
+        AIRBYTE_PLATFORM_INTERNAL_GITHUB_REPO_URL, local_repo_path, branch=AIRBYTE_PLATFORM_INTERNAL_MAIN_BRANCH_NAME
+    )
```
Comment (evantahler) [inline tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py:32]: Yeah, I guess we have to clone the platform repo to open a PR

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py [hunk 4]
```diff
@@ -50,31 +61,64 @@ def update_definitions_mask(connector: ConnectorQAReport, definitions_mask_path:
...
+    return len(response.json()) > 0
+
+
+def create_pr(connector: ConnectorQAReport, branch: str) -> Optional[requests.Response]:
+    body = f"""The Cloud Availability Updater decided that it's the right time to make {connector.connector_name} available on Cloud!
+    Technical name: {connector.connector_technical_name}
+    Version: {connector.connector_version}
+    Definition ID: {connector.connector_definition_id}
+    OSS sync success rate: {connector.sync_success_rate}
+    OSS number of connections: {connector.number_of_connections}
```
Comment (evantahler) [inline tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py:103]: Nit: Maybe the body should have markdown bullet points?

**Review by evantahler — APPROVED** [review]

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py (hunk 2 above)
Comment (alafanechere) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py:2]: The autoformatting is still in 2022 :(

Comment (alafanechere) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py:2]: https://github.com/airbytehq/airbyte/pull/22576 updated it :)

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py (hunk 4 above)
Comment (alafanechere) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py]: Done

**Pull request merged**